clang/lib/AST/DeclCXX.cpp

```diff
@@ -1696,7 +1696,11 @@ static NamedDecl* getLambdaCallOperatorHelper(const CXXRecordDecl &RD) {
       RD.getASTContext().DeclarationNames.getCXXOperatorName(OO_Call);
 
   DeclContext::lookup_result Calls = RD.lookup(Name);
-  assert(!Calls.empty() && "Missing lambda call operator!");
+
+  // This can happen while building the lambda.
+  if (Calls.empty())
+    return nullptr;
```

Comment:
So I'm REALLY concerned about this.  I don't think it is wrong, but just that we need to make sure we handle this value correctly.  At least `getLambdaStaticInvoker` doesn't handle the `nullptr` correctly (though I guess the other two uses DO, so thats fine).

I wouldn't mind a bit of an audit to make sure that uses of `getLambdaStaticInvoker` (and perhaps the others, but as they already acknowledged they could return null, i'm mildly less concerned) do a reasonable thing.

Reply:
I added an assert, which hinges on the fact that we don't look at the static invoker in cases where the lambda is incomplete.
I agree that having these various states in lambdas is a bit unfortunate, but it is a consequence of P2036.
If you have a better solution, I'm all ears!